csharp/src/Google.Protobuf/Collections/RepeatedField.cs

```diff
@@ -101,10 +101,11 @@ public void AddEntriesFrom(CodedInputStream input, FieldCodec<T> codec)
                 int length = input.ReadLength();
                 if (length > 0)
                 {
+                    EnsureSize(count + length);
```

Comment:
Thanks a lot for your review. You are definitely right. It was because of my misundesrtanding of the pack format. Any performance improvement at this point would require some guessing or approximation and does not seem to be feasible..

This also renders my idea of allowing using structs instead of classes useless because of high cost of reallocations. That's disapointing. I wanted to allocate a contiguous array of structs at this point what should be way more GC friendly in many common scenarios.

I will stick to the new no-GC region feature - if used correctly it should give comparable results.

However, I have different idea of optimization in scenarios where protobuf generated parser generates temp representation of data - hybrid approach allowing using streaming for some repeated fields (decorated with some options).